### PR TITLE
Adds e2e image publishing into CI pipelines

### PR DESCRIPTION
**Pull request by m1kola**
### Which issue this PR addresses:

Fixes [Work Item#6602294](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/6602294)

### What this PR does / why we need it:

Preparation to CI refactoring so we can use a pre-built e2e test binary of specific version against the RP of the same version.

### Test plan for issue:

```
export RP_IMAGE_ACR=test-acr
make image-aro
```

Also need to run e2e in the PR itself to make sure that e2e did not break due to make target changes.

### Is there any documentation that needs to be updated for this PR?

Documentation updated as part of this PR (make target changes), but still a good idea to notify the team about changes.

### CODE REVIEW

**Comment by mjudeikis** [comment]
@m1kola why not just add the image into the same image as RP binary? and have 2 binaries in the same image?

**Comment by m1kola** [comment]
@mjudeikis I thought about it. We can do that, but there are few (minor) reasons:

* It will increase RP image size a bit (so slightly longer download time for prod deployments)
* Not sure if we want to have the e2e binary in prod. I do not see any attack vector or anything like that, but want to keep prod image minimal and separate to test stuff

Do you think it worth having a single image?

**Comment by mjudeikis** [comment]
@jim-minter, what is your take on this? Separate image or "one image, two binaries?"

**Comment by mjudeikis** [comment]
@jim-minter what is your take on this? two images or one image with 2 binaries?

**Comment by jim-minter** [comment]
I prefer one image with two binaries.

File: Makefile [hunk 1]
```diff
@@ -110,7 +110,10 @@ secrets-update:
 	rm secrets.tar.gz
 
 e2e:
-	go test ./test/e2e -timeout 60m -v -ginkgo.v -tags e2e
+	go test ./test/e2e -tags e2e -c -o e2e.test
+
+test-e2e: e2e
```
Comment (jim-minter) [inline Makefile:115]: How about calling the previous target e2e.test, given that's the name of the binary it actually creates?  Avoids changing the meaning of `make e2e`.

Comment (m1kola) [reply Makefile:115]: Yeah, I thought about it, but I think it is going to be confusing to have both `e2e` and `e2e.test`. Also if you were to see these targets for the first time, which one is actually running tests? I think it makes sense to standartise on `test-*` for targets that run tests (we already have `test-go`, `test-python`). I'll just send an email to the team about changes in the target.

Comment (jim-minter) [reply Makefile:115]: `test-e2e` makes sense, I agree. Still tempted to call the previous target e2e.test as that's the file it creates.

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 SHELL = /bin/bash
 COMMIT = $(shell git rev-parse --short HEAD)$(shell [[ $$(git status --porcelain) = "" ]] || echo -dirty)
 
-aro: generate
+aro: generate e2e
```
Comment (jim-minter) [inline Makefile:4]: shouldn't this change go on the image-aro target?

**Comment by mjudeikis** [comment]
Only these 2 nits (as per Jims comments) and we can merge.
1. target rename `e2e.test` to match `admin.kubeconfig` too.
2. Move `e2e.test` target to `aro-image`

**Comment by m1kola** [comment]
VM redeploymnet e2e (not related to this PR failed. Re-running to see if it was a flake.

/azp run e2e

**Comment by m1kola** [comment]
@mjudeikis @jim-minter I rebased and fixed conflicts. Also addressed the feedback. There was an e2e failure in VM redeploy, but it looks like it was an Azure flake (API returned 500). Please take another look.

**Pull request merged**